### ISSUE TITLE
[LAB] Week 1_Problem 1_Dongwahn Kim

### ISSUE THREAD

**Issue by dongwahn78** [vlab-kaist/NN101_23S#16]
### Week

Week 1_Problem 1

### Source Code

```python
import torch
from random import random 
from typing import Callable


def solution(func: Callable, start_point: float) -> float:
    x = torch.tensor(start_point, requires_grad=True)
    y = func(x)
    while True:
        grad, = torch.autograd.grad(y, x, create_graph=True)
        x = x - 0.1 * grad
        y = func(x)
        if grad.abs() < 1e-5:
            break
    return x.item()


if __name__ == "__main__":
    def test_func(x):
        return x ** 2
    t = 10*random()
   



### Description

done

### Output (Optional)

_No response_

**Comment by seo-rii** [comment]
Please change issue's title correctly!

**Comment by jangyoujin0917** [comment]
Please do not use optim in PyTorch.
This problem is for checking your understanding for gradient-based optimization.

**Comment by Dongyeongkim** [comment]
This issue is closed now because of the lack of progress.
